### DISCUSSION TITLE
Version 1

### DISCUSSION THREAD

**Discussion by mawoka-myblock** [mawoka-myblock/ClassQuiz#92] [General]
Yea, this will probably take it's time.

I want to know the following things before releasing version 1:
- How many instances of ClassQuiz are around?
- Is self-hosting a priority?
- What do the users want?
- Is ClassQuiz stable (I mean, _of course, it is_ 😉 )
- How will I not loose interest in ClassQuiz?
- What will happen if I have less and less time to work on ClassQuiz?
- Which features (can) make ClassQuiz unique?
- How can I test new features better (maybe a dev-instance)?
- What happens if I add a new feature, which breaks an old one and I am not interested to fix the old one (eg. The range-answers broke the export-function)
- How can I build a (small) community around this project?

@Dynnammo @Cittjr

**Already, thanks to all the awesome people who helped to this day.**

**Comment by Dynnammo** [comment]
Many questions to answer :cold_sweat: .
> How many instances of ClassQuiz are around? Is self-hosting a priority?

Do not know. Since the main website is functional (so no need to self-host yet)and the project still early, I do not think there's much. I think the second question is linked to the first : not in my opinion for the reasons invoked above.

> What do the users want?

Good question. Since users of quiz-tools are mainly teachers/students, I do not think they are acquainted with GitHub or

> Is ClassQuiz stable (I mean, of course, it is wink )

More and more everyday :smile:.

> How will I not loose interest in ClassQuiz? What will happen if I have less and less time to work on ClassQuiz?

- Having other benevolants to help push the project forward. 
- Not working for free i.e. winning a bit of money what you do : developing new features / paying for the hosting of the website + tools

> Which features (can) make ClassQuiz unique? How can I test new features better (maybe a dev-instance)?

Having features competitors do not have :smile: : graphically customizable, offer a wider range of question types (maps to click one, range etc.). Completely agree on the last point.

> What happens if I add a new feature, which breaks an old one and I am not interested to fix the old one (eg. The range-answers broke the export-function)

The purpose of a test cover is precisely to avoid such behavior : if you develop a new feature that breaks an old one, the CI prevents you to push it in production (or it pushes it at least not on the main website).

> How can I build a (small) community around this project?

**Involving developers/designers**
Taking part in hackaton/events of code (Hacktoberfest of this year, hackaton about education). Personaly, I would happily contribute for this part

**Involving teachers/students**
- Advertising by _word of mouth_ teachers/communities of teachers of colleges/universities we each know to get **beta-testers** or people willing to comment/ask for new features
- These same people can be retained through more "muggles tools" like Discourse or any chat app that are relevant
